activity-monitor.js

```diff
@@ -103,13 +118,35 @@
                 if (service.user.active && service.user.action <= inactive) {
                     service.user.active = false;
                     publish(EVENT_INACTIVE);
-                    disable();
+
+                    if(service.options.disableOnInactive){
+                        disable();
+                    }else{
+                        disableIntervals();//user inactive is known, lets stop checking, for now
+                        dynamicActivity = reactivate;//hot swap method that handles document event watching
+                    }
                 }
             }, service.options.monitor * MILLISECOND);
         }
 
+        /* function that lives in memory with the intention of being swapped out */
+        function dynamicActivity(){
+            regularActivityMonitor();
+        }
+
+        /* after user inactive, this method is hot swapped as the dynamicActivity method in-which the next user activity reactivates monitors */
+        function reactivate() {
+            enableIntervals();
+            dynamicActivity = regularActivityMonitor;
+        }
+
         /* invoked on every user action */
-        function activity() {
+        function activity(){
+            dynamicActivity()
```

Comment:
Fair enough, just running through all scenarios :)
Admittingly, I haven't touched this code in awhile haha

I'll merge this and publish/release a little later today. I'd like to solve #6 before the next release.